src/main/java/com/conveyal/r5/transit/TransitLayer.java

```diff
@@ -723,8 +721,14 @@ public static TransitModes getTransitModes(int routeType) {
             return TransitModes.GONDOLA;
         case 7:
             return TransitModes.FUNICULAR;
+        // Codes 8, 9, and 10 are unused.
+        // Codes 11 and 12 were officially added to the spec to bring it in line with internal Google usage.
+        case 11:
+            return TransitModes.BUS;  // Trolleybus
+        case 12:
+            return TransitModes.RAIL; // Monorail
```

Comment:
Let's discuss - I'd be inclined to group this with METRO instead of RAIL (which we typically use for long-distance/regional rail).

Reply:
`SUBWAY` could work. I didn't think through the RAIL choice much, believing the monorail code wouldn't be used much and wouldn't really influence our output. Looking at the GTFS spec, subway/metro is defined as being underground. Another possible option is `0 - Tram, Streetcar, Light rail. Any light rail or street level system within a metropolitan area` but I'm not sure if this is strictly for at-grade systems. "Light rail" is sometimes elevated e.g. in Singapore, and the formulation "any light rail _or_ street level system" implies that light rail is often  _not_ at street level. All monorails I'm familiar with serve a comparable function to the Singpore light rail (e.g. Punggol LRT).

Reply:
I think the GTFS spec definition for `SUBWAY` should be revised grade-separated rather than underground, but given the current definition, I'd vote for `TRAM`.

Reply:
Good examples, those are definitely monorails as part of a metro system. I should have thought of the Chongqing case. The main problem here is that "monorail" is a guideway or traction technology, not a mode of transportation in the sense "metro" is (grade-separated high capacity). This is a bit like having a separate mode code for "rubber tires" which could then apply to a subset of the Paris Metro lines. 

We should suggest the change from "underground" to "grade-separated" in the GTFS spec if it hasn't already been proposed and/or rejected.

Reply:
Hmmm. Maybe I forgot the original reasoning and conversation around this, but it looks like we're essentially mapping 11 to 3 and 12 to 2 here, which does not feel right at all...

Why wouldn't we allow trolleybus and monorail to be their own types? 
If it's a usability issue, shouldn't this be addressed in the UI?

Reply:
Some background:

R5 has always (since 2015) included `TransitMode` enums corresponding to GTFS `route_type` values 0-7 (see history of https://github.com/conveyal/r5/blob/v6.5/src/main/java/com/conveyal/r5/api/util/TransitModes.java). Filtering for these modes was added to R5 in https://github.com/conveyal/r5/pull/355 and (in 2017), and to the UI in 529 (in 2017, for `route_type` 0-3) and 776 (in 2018, for `route_type` 4-7).

In 2020, the GTFS spec was amended to add `route_type` values 11 and 12 (via https://github.com/google/transit/pull/174). My initial thought was to update the `TransitMode` enums in R5 accordingly, but after reviewing the timeline, I agree with @abyrd's assessment about the potential disruptiveness of doing so. While the mapping approach could require additional debugging effort in the future, we already do the mapping of TPEG modes, and already lack a 1:1 mapping of `TransitMode` to GTFS `route_type` (thanks to `AIR`, though that was admittedly a more recent change)

Reply:
```suggestion
            return TransitModes.TRAM; // Monorail
```